modules/gdscript/gdscript_warning.h

```diff
@@ -146,6 +147,7 @@ class GDScriptWarning {
 		ERROR, // NATIVE_METHOD_OVERRIDE // May not work as expected.
 		ERROR, // GET_NODE_DEFAULT_WITHOUT_ONREADY // May not work as expected.
 		ERROR, // ONREADY_WITH_EXPORT // May not work as expected.
+		ERROR, // INIT_WITH_REQUIRED_PARAM // Can prevent objects from being correctly instantiated or duplicated
```

Comment:
```suggestion
		WARN, // INIT_WITH_REQUIRED_PARAM
```

I don't think this should be treated as an error by default due to compatibility reasons. I think a warning would be sufficient.